driver-core/src/test/java/com/datastax/driver/core/CustomTypeTest.java

```diff
@@ -22,26 +22,85 @@
 import java.util.List;
 
 import org.testng.annotations.Test;
-import static org.testng.Assert.assertEquals;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+import static com.datastax.driver.core.Assertions.assertThat;
+import static com.datastax.driver.core.DataType.cint;
+import static com.datastax.driver.core.DataType.custom;
+import static com.datastax.driver.core.DataType.list;
 
 /**
  * Test we "support" custom types.
  */
 public class CustomTypeTest extends CCMBridge.PerClassSingleNodeCluster {
 
+    public static final DataType CUSTOM_DYNAMIC_COMPOSITE = custom(
+        "org.apache.cassandra.db.marshal.DynamicCompositeType("
+        + "s=>org.apache.cassandra.db.marshal.UTF8Type,"
+        + "i=>org.apache.cassandra.db.marshal.Int32Type)");
+
+    public static final DataType CUSTOM_COMPOSITE = custom(
+        "org.apache.cassandra.db.marshal.CompositeType("
+            + "org.apache.cassandra.db.marshal.UTF8Type,"
+            + "org.apache.cassandra.db.marshal.Int32Type)");
+
+    public static final DataType LIST_OF_INTS = list(cint());
+
     @Override
     protected Collection<String> getTableDefinitions() {
         return Collections.singleton(
-              "CREATE TABLE test ("
-            + "    k int,"
-            + "    c 'DynamicCompositeType(s => UTF8Type, i => Int32Type)',"
-            + "    v int,"
-            + "    PRIMARY KEY (k, c)"
-            + ") WITH COMPACT STORAGE"
+            "CREATE TABLE test ("
+                + "    k int,"
+                + "    c1 'DynamicCompositeType(s => UTF8Type, i => Int32Type)',"
+                + "    c2 'ReversedType(CompositeType(UTF8Type, Int32Type))'," // reversed translates to CLUSTERING ORDER BY DESC
+                + "    c3 'ListType(Int32Type)'," // translates to list<int>
```

Comment:
That works for me :+1:.  It turns out the problem wasn't ListType being used with c1 and c2.  It's just list type in general with 1.2.  When I tested it alone I still got the same failure, it seems C* 1.2 can't figure out what to do with the custom type on the table declaration, but it works fine as 'list<int>' (as we'd expect).

I made a [change](https://gist.github.com/tolbertam/26c0c4f28f782ef61d1a) locally to use a different type for 1.2, but I think your solution is more straightforward and elegant.